include/irr/asset/filters/CSwizzleAndConvertImageFilter.h

```diff
@@ -151,10 +164,37 @@ class CSwizzleAndConvertImageFilter : public CImageFilter<CSwizzleAndConvertImag
 					{
 						auto localOutPos = readBlockPos*blockDims+commonExecuteData.offsetDifference;
 						uint8_t* dstPix = commonExecuteData.outData+commonExecuteData.oit->getByteOffset(localOutPos,commonExecuteData.outByteStrides);
-						if constexpr(std::is_base_of<PolymorphicSwizzle,Swizzle>::value)
-							convertColor<inFormat,outFormat,Swizzle>(srcPix,dstPix,blockX,blockY,state->swizzle);
-						else
-							convertColor<inFormat,outFormat,Swizzle>(srcPix,dstPix,blockX,blockY,*state);
+						
+						auto getSwizzle = [&]() -> Swizzle&
+						{
+							if constexpr (std::is_base_of<PolymorphicSwizzle, Swizzle>::value)
+								return state->swizzle;
+							else
+								return *state;
+						};
+
+						const Swizzle& swizzle = getSwizzle();
+
+						constexpr auto maxChannels = 4;
+						uint64_t decodeBuffer[maxChannels] = {};
+						uint64_t encodeBuffer[maxChannels] = {};
+
+						asset::decodePixels<inFormat>(srcPix, reinterpret_cast<decodeBufferType*>(decodeBuffer), blockX, blockY);
+						swizzle.template operator()<void, void>(decodeBuffer, encodeBuffer);
+
+						auto clampBuffer = [&]()
+						{
+							for (uint8_t i = 0; i < clampBufferChannelsAmount; ++i)
+							{
+								auto&& [min, max, encodeValue] = std::make_tuple<encodeBufferType&&, encodeBufferType&&, encodeBufferType*>(asset::getFormatMinValue<encodeBufferType>(outFormat, i), asset::getFormatMaxValue<encodeBufferType>(outFormat, i), reinterpret_cast<encodeBufferType*>(encodeBuffer) + i);
+								*encodeValue = core::clamp(*encodeValue, min, max);
+							}
+						};
+
+						if constexpr (clamp)
+							clampBuffer();
+							
+						asset::encodePixels<outFormat>(dstPix, reinterpret_cast<encodeBufferType*>(encodeBuffer));
```

Comment:
as I think of it, shouldn't I move a pointer to be a `dstPix+= blockSize` each iteration in block loop?

Reply:
okay, we do not support encoding to BC, it clears a little

Reply:
should still advance the pointer though XD ;)

Reply:
yup, as well as I do it on SAT now

Reply:
corrected in following commit